File: js/6550.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const ans = [];
inputs.forEach((input) => {
  const [s, t] = input.split(' ');

  let idx = 0;

  for (let i = 0; i < t.length; i++) {
    if (t[i] === s[idx]) idx++;
  }

  ans.push(idx === s.length ? 'Yes' : 'No');
});

console.log(ans.join('\n'));

File: c++/6550.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);

  string s, t;
  while (cin >> s >> t) {
    int idx = 0;

    for (int i = 0; i < t.length(); i++) {
      if (t[i] == s[idx]) {
        idx++;
      }
    }

    if (idx == s.length())
      cout << "Yes" << '\n';
    else
      cout << "No" << '\n';
  }

  return 0;
}